openfe/protocols/openmm_septop/equil_septop_method.py

```diff
@@ -1802,13 +1796,17 @@ def run(
             )
         )
 
+        smc_off_B = {m: m.to_openff() for m in alchem_comps["stateB"]}
```

Comment:
Recreating the openff molecule creates a separate object, which is not what you want to do.

What you want to do here is find the keys that belong in `smc_comps_B` but not in `smc_comps_A`.

So you can do something like:

```suggestion
        smc_B_unique_keys = smc_comps_B.keys() - smc_comp_A.keys()
        smc_comp_B_unique = {key: smc_comps_B[key] for key in smc_B_unique_keys}
        omm_system_AB, omm_topology_AB, positions_AB, modeller_AB = self.get_system_AB(
            solv_comp,
            modeller_A,
            smc_comps_AB,
            smc_comp_B_unique,
            settings,
        )
```

P.S. renaming from smc_off_B to smc_compp_B_unique, because the naming is confusing compared to smc_comp_B, etc...